test/sql/distinct_sql_test.cpp

```diff
@@ -0,0 +1,161 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// distinct_sql_test.cpp
+//
+// Identification: test/sql/distinct_sql_test.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <memory>
+
+#include "catalog/catalog.h"
+#include "common/harness.h"
+#include "executor/create_executor.h"
+#include "optimizer/simple_optimizer.h"
+#include "planner/create_plan.h"
+#include "sql/sql_tests_util.h"
+
+namespace peloton {
+namespace test {
+
+class DistinctSQLTests : public PelotonTest {};
+
+void CreateAndLoadTable() {
+  // Create a table first
+  SQLTestsUtil::ExecuteSQLQuery(
+      "CREATE TABLE test(a INT PRIMARY KEY, b INT, c INT, d VARCHAR);");
+
+  // Insert tuples into table
+  SQLTestsUtil::ExecuteSQLQuery(
+      "INSERT INTO test VALUES (1, 22, 333, 'abcd');");
+  SQLTestsUtil::ExecuteSQLQuery(
+      "INSERT INTO test VALUES (2, 22, 333, 'abc');");
+  SQLTestsUtil::ExecuteSQLQuery(
+      "INSERT INTO test VALUES (3, 11, 222, 'abcd');");
+}
+
+TEST_F(DistinctSQLTests, DistinctIntTest) {
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, nullptr);
+
+  CreateAndLoadTable();
+
+  std::vector<StatementResult> result;
+  std::vector<FieldInfo> tuple_descriptor;
+  std::string error_message;
+  int rows_changed;
+
+  SQLTestsUtil::ExecuteSQLQuery("SELECT DISTINCT b FROM test;", result,
+                                tuple_descriptor, rows_changed, error_message);
+
+  // Check the return value
+  // Should be: 22, 11
+  EXPECT_EQ(0, rows_changed);
```

Comment:
Can you check the result size as you did in DistinctStarTest? The same for DistinctVarcharTest and DistinctTupleTest.